packages/x-charts/src/ChartsAxisHighlight/ChartsAxisHighlight.tsx

```diff
@@ -80,11 +80,12 @@ function ChartsAxisHighlight(props: ChartsAxisHighlightProps) {
 
   const getXPosition = getValueToPositionMapper(xScale);
   const getYPosition = getValueToPositionMapper(yScale);
+
   return (
     <React.Fragment>
       {xAxisHighlight === 'band' && axis.x !== null && isBandScale(xScale) && (
         <ChartsAxisHighlightPath
-          d={`M ${xScale(axis.x.value)! - (xScale.step() - xScale.bandwidth()) / 2} ${
+          d={`M ${(xScale(axis.x.value) ?? 0) - (xScale.step() - xScale.bandwidth()) / 2} ${
```

Comment:
If `xScale(axis.x.value)` is not defined, Maybe it would make more sense not to render this component instead of placing it at coordinate 0

Reply:
I went with rendering at 0 because it would be clearer that something is wrong, rather than silently failing. Though I don't have a strong preference

Reply:
What about logging a warning?

This should not append, but I think it's better to have a non working features that logs in the console why it's not working

Reply:
🤔 Wait I did something wrong 😅

Reply:
fixed 👍